src/lib/flags.js

```diff
@@ -0,0 +1,47 @@
+let flags
+
+/**
+ * Read CLI flags and populate userland options
+ */
+module.exports = function getFlags () {
+  if (flags) return flags
+
+  // TODO refactor all this janky custom logic into something like yargs
+  let args = process.argv.slice(2)
+
+  // Log levels
+  let logLevel = 'normal'
+  let findVerbose = option => [ '-v', '--verbose', 'verbose' ].includes(option)
+  let findDebug =   option => [ '-d', '--debug', 'debug' ].includes(option)
+  if (args.some(findVerbose)) logLevel = 'verbose'
+  if (args.some(findDebug)) logLevel = 'debug'
+
+  // Get the base port for HTTP / WebSockets; tables + events key off this
+  // CLI args > env var
+  let port = Number(process.env.PORT) || 3333
+  let findPort = option => [ '-p', '--port', 'port' ].includes(option)
+  if (args.some(findPort)) {
+    let thePort = i => args[args.indexOf(i) + 1]
+    if (args.includes('-p'))           port = thePort('-p')
+    else if (args.includes('--port'))  port = thePort('--port')
+    else if (args.includes('port'))    port = thePort('port')
+    if (isNaN(Number(port)))           port = 3333
+  }
+
+  // Quiet stdout
+  let findQuiet = option => [ '-q', '--quiet', 'quiet' ].includes(option)
+  let quiet = args.some(findQuiet)
+  // TODO this is also janky af but we'll deal for now
```

Comment:
Yeah I'm not a huge fan of what looks like two variables being used to track verbosity (quiet and logLevel) - and the updater PR does this same thing actually. Would be nice to consolidate down the road. The updater PR, too, looks at environment variables to determine whether `quiet` is set or not. Think ultimately we'll want to have the arc child packages parse env vars themselves and not have `utils` do it.

Reply:
The thinking here was: logLevel sets the amount of data that's logged (and is accessible when these are used as modules and not as CLI tools), while quiet suppressed stdout. I've found this to be useful and necessary when calling Arc modules outside the CLI.

I think the extra env var setting here was just because at the time I wrote this there were a bunch of extra updater instantiations still in the codebase, will see if I can tidy it up, thanks!